diff --git a/data/com.github.wwmm.pulseeffects.appdata.xml.in b/data/com.github.wwmm.pulseeffects.appdata.xml.in
index 7bdf27e..25e8b3e 100644
--- a/data/com.github.wwmm.pulseeffects.appdata.xml.in
+++ b/data/com.github.wwmm.pulseeffects.appdata.xml.in
@@ -57,5 +57,5 @@
     </screenshot>
   </screenshots>
   <update_contact>contact_AT_hembas.se</update_contact>
-  <translation type="gettext">pulseeffects</translation>
+  <translation type="gettext">PulseEffects</translation>
 </component>
diff --git a/data/meson.build b/data/meson.build
index f9f03dc..5da01bb 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -1,4 +1,4 @@
-schemadir = join_paths(get_option('datadir'), 'glib-2.0/schemas')
+schemadir = join_paths(datadir, 'glib-2.0', 'schemas')
 install_data([
   'com.github.wwmm.pulseeffects.gschema.xml',
   'com.github.wwmm.pulseeffects.sinkinputs.gschema.xml',
@@ -10,7 +10,7 @@ i18n_mod.merge_file(
   output: 'com.github.wwmm.pulseeffects.appdata.xml',
   po_dir: '../po',
   install: true,
-  install_dir: join_paths(get_option('datadir'), 'metainfo')
+  install_dir: join_paths(datadir, 'metainfo')
 )
 
 i18n_mod.merge_file(
@@ -19,10 +19,10 @@ i18n_mod.merge_file(
   type: 'desktop',
   po_dir: '../po',
   install: true,
-  install_dir: join_paths(get_option('datadir'), 'applications')
+  install_dir: join_paths(datadir, 'applications')
 )
 
-icondir = join_paths(get_option('datadir'), 'icons/hicolor')
+icondir = join_paths(datadir, 'icons', 'hicolor')
 install_data('pulseeffects.svg',
-  install_dir: join_paths(icondir, 'scalable/apps')
+  install_dir: join_paths(icondir, 'scalable', 'apps')
 )
diff --git a/meson.build b/meson.build
index 733196a..1d3681a 100644
--- a/meson.build
+++ b/meson.build
@@ -8,13 +8,25 @@ i18n_mod = import('i18n')
 message('Checking dependencies')
 py3 = py3_mod.find_python()
 
+prefix = get_option('prefix')
+bindir = join_paths(prefix, get_option('bindir'))
+datadir = join_paths(prefix, get_option('datadir'))
+localedir = join_paths(prefix, get_option('localedir'))
+
 message('Getting python install path')
 py3_dir = py3_mod.sysconfig_path('purelib')
 
 message('Installing PulseEffects')
 install_subdir('PulseEffects', install_dir: py3_dir)
 install_subdir('PulseEffectsCalibration', install_dir: py3_dir)
-install_data('pulseeffects', install_dir: 'bin')
+
+conf_pulseeffects = configuration_data()
+conf_pulseeffects.set('LOCALEDIR', localedir)
+conf_pulseeffects.set('PROJECT_NAME', meson.project_name())
+configure_file(input: 'pulseeffects.in',
+               output: 'pulseeffects',
+               configuration: conf_pulseeffects,
+               install_dir: bindir)
 
 message('Installing data files')
 subdir('data')
diff --git a/pulseeffects b/pulseeffects
deleted file mode 100755
index 3316849..0000000
--- a/pulseeffects
+++ /dev/null
@@ -1,10 +0,0 @@
-#!/usr/bin/env python3
-
-import sys
-from PulseEffects.application import Application
-
-a = Application()
-
-exit_status = a.run(sys.argv)
-
-sys.exit(exit_status)
diff --git a/pulseeffects.in b/pulseeffects.in
new file mode 100755
index 0000000..c1466be
--- /dev/null
+++ b/pulseeffects.in
@@ -0,0 +1,17 @@
+#!/usr/bin/env python3
+
+import gettext
+import locale
+import sys
+from PulseEffects.application import Application
+
+gettext.bindtextdomain('@PROJECT_NAME@', '@LOCALEDIR@')
+gettext.textdomain('@PROJECT_NAME@')
+locale.bindtextdomain('@PROJECT_NAME@', '@LOCALEDIR@')
+locale.textdomain('@PROJECT_NAME@')
+
+a = Application()
+
+exit_status = a.run(sys.argv)
+
+sys.exit(exit_status)
