diff -urN a/Makefile b/Makefile
--- a/Makefile	2017-12-23 11:59:57.249231089 +0100
+++ b/Makefile	2017-12-23 13:45:22.861830306 +0100
@@ -5,7 +5,7 @@
 NAME    = zam-plugins
 VERSION = $(shell cat .version)
 
-PLUGINS=ZamComp ZamCompX2 ZaMultiComp ZamTube ZamEQ2 ZamAutoSat ZamGEQ31 ZaMultiCompX2 ZamGate ZamGateX2 ZamHeadX2 ZaMaximX2 ZamDelay ZamDynamicEQ ZamPhono
+PLUGINS=ZaMaximX2
 #ZamPiano ZamSFZ ZamChild670
 
 include Makefile.mk
