From d33c0dbb4c7e07076865b187ccb58cb2c4da1c74 Mon Sep 17 00:00:00 2001
From: Markus Schmidt <schmidt@boomshop.net>
Date: Wed, 20 Jun 2018 17:09:35 +0200
Subject: [PATCH] Limiter: try to fix memleak

---
 src/audio_fx.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/audio_fx.cpp b/src/audio_fx.cpp
index 98a368fc..d2a8dfbd 100644
--- a/src/audio_fx.cpp
+++ b/src/audio_fx.cpp
@@ -643,6 +643,11 @@ float lookahead_limiter::get_attenuation()
 void lookahead_limiter::set_sample_rate(uint32_t sr)
 {
     srate = sr;
+    
+    free(buffer);
+    free(nextpos);
+    free(nextdelta);
+    
     // rebuild buffer
     overall_buffer_size = (int)(srate * (100.f / 1000.f) * channels) + channels; // buffer size attack rate multiplied by 2 channels
     buffer = (float*) calloc(overall_buffer_size, sizeof(float));
-- 
2.17.1

