diff -urN a/Makefile b/Makefile
--- a/Makefile	2018-01-26 21:44:10.471163537 +0100
+++ b/Makefile	2018-02-07 17:37:15.452336504 +0100
@@ -1,6 +1,6 @@
-BIN_PATH                = /usr/local/bin
-LIB_PATH                = /usr/local/lib
-DOC_PATH                = /usr/local/share/doc
+BIN_PATH                = /app/bin
+LIB_PATH                = /app/lib
+DOC_PATH                = /app/share/doc
 LADSPA_PATH             = $(LIB_PATH)/ladspa
 LV2_PATH                = $(LIB_PATH)/lv2
 VST_PATH                = $(LIB_PATH)/vst
@@ -26,12 +26,9 @@
 export RELEASE          = ${CURDIR}/.release
 export BUILDDIR         = $(OBJDIR)
 
-# Dependencies
-export VST_SDK          = /home/sadko/eclipse/lsp-plugins-vst3sdk
-
 # Includes
 INC_FLAGS               = -I"${CURDIR}/include"
-INSTALLATIONS           = install_ladspa install_lv2 install_jack install_doc
+INSTALLATIONS           = install_lv2
 RELEASES                = release_ladspa release_lv2 release_jack release_src release_doc
 ifdef VST_SDK
 INC_FLAGS              += -I"$(VST_SDK)"
@@ -108,8 +105,6 @@
 export EXPAT_LIBS       = $(shell pkg-config --libs expat)
 export SNDFILE_HEADERS  = $(shell pkg-config --cflags sndfile)
 export SNDFILE_LIBS     = $(shell pkg-config --libs sndfile)
-export JACK_HEADERS     = $(shell pkg-config --cflags jack)
-export JACK_LIBS        = $(shell pkg-config --libs jack)
 
 FILE                    = $(@:$(OBJDIR)/%.o=%.cpp)
 FILES                   =
@@ -176,7 +171,6 @@
 	@echo "Installing LV2 plugins to $(DESTDIR)$(LV2_PATH)/$(ARTIFACT_ID).lv2"
 	@mkdir -p $(DESTDIR)$(LV2_PATH)/$(ARTIFACT_ID).lv2
 	@$(INSTALL) $(LIB_LV2) $(DESTDIR)$(LV2_PATH)/$(ARTIFACT_ID).lv2/
-	@$(INSTALL) $(LIB_LV2_GTK2UI) $(DESTDIR)$(LV2_PATH)/$(ARTIFACT_ID).lv2/
 	@$(UTL_GENTTL) $(DESTDIR)$(LV2_PATH)/$(ARTIFACT_ID).lv2
 	
 install_vst: all
diff -urN a/src/container/Makefile b/src/container/Makefile
--- a/src/container/Makefile	2018-01-26 21:44:10.457163653 +0100
+++ b/src/container/Makefile	2018-02-07 17:38:53.470646837 +0100
@@ -4,7 +4,7 @@
 OBJ_HELPERS             = $(OBJDIR)/CairoCanvas.o
 
 # Detemine what modules to build
-MODULES                 = $(LIB_LADSPA) $(LIB_LV2) $(LIB_LV2_GTK2UI) $(LIB_JACK) $(BIN_PROFILE)
+MODULES                 = $(LIB_LV2)
 ifdef VST_SDK
 MODULES                += $(LIB_VST)
 endif
diff -urN a/src/Makefile b/src/Makefile
--- a/src/Makefile	2018-01-26 21:44:10.455163670 +0100
+++ b/src/Makefile	2018-02-07 17:40:15.563556637 +0100
@@ -1,5 +1,5 @@
-SUBDIRS                 = core metadata plugins ui utils container doc
-MODULES                 = $(SUBDIRS) jack_stub
+SUBDIRS                 = core metadata plugins utils container
+MODULES                 = $(SUBDIRS)
 
 ifdef VST_SDK
 MODULES                += vst_stub
